File: cryptoscope/crates/scan-source/tests/fixtures/brokenclassical/Legacy.java
// Broken-classical Java call sites. Every one of these is a textbook finding
// that the scanner reported as zero before the reachability work: RC4 and
// Signature.getInstance had no matcher at all, and DESede was misattributed
// to single DES.
package fixtures;

import javax.crypto.Cipher;
import java.security.Signature;

public class Legacy {
    // CRYPTO-214 — 3DES, not DES.
    public static Cipher tripleDes() throws Exception {
        return Cipher.getInstance("DESede/CBC/PKCS5Padding");
    }

    // CRYPTO-213 — RC4.
    public static Cipher rc4() throws Exception {
        return Cipher.getInstance("RC4");
    }

    // CRYPTO-291 — SHA-1 signature.
    public static Signature sha1WithRsa() throws Exception {
        return Signature.getInstance("SHA1withRSA");
    }
}

File: cryptoscope/crates/scan-source/tests/fixtures/brokenclassical/legacy.go
// Broken-classical Go stdlib call sites. crypto/des and crypto/rc4 exist only
// for legacy interop; neither had a rule before, and crypto/aes.NewCipher had
// a rule with no matcher to feed it.
package brokenclassical

import (
	"crypto/aes"
	"crypto/cipher"
	"crypto/des"
	"crypto/rc4"
)

// CRYPTO-042
func TripleDES(key []byte) (cipher.Block, error) {
	return des.NewTripleDESCipher(key)
}

// CRYPTO-043
func RC4(key []byte) (*rc4.Cipher, error) {
	return rc4.NewCipher(key)
}

// CRYPTO-040
func AES(key []byte) (cipher.Block, error) {
	return aes.NewCipher(key)
}
